cmd/snap-mgmt/snap-mgmt.sh.in

```diff
@@ -119,6 +119,28 @@ purge() {
         rm -f "/etc/systemd/system/multi-user.target.wants/$unit"
     done
 
+    # TODO: ask snapd to ask each snapd.session-agent.service to stop snaps
+    # user-session services and stop itself.
+
+    user_units=$(systemctl --user --global list-unit-files --no-legend --full)
+    # user services from snaps
+    user_services=$(echo "$user_units" | grep '^snap\..*\.service' | cut -f1 -d ' ')
+    for unit in $user_services; do
```

Comment:
I think this should be done in the for loop above, under the `if [ -n "$snap" ]; then` in line 79. The user units share the same `snap.${snap}` prefix so we can drop all of them there.

Reply:
Ah, so tying the cleanup code to the presence of the mount unit? Yeah, I guess that is simpler. I'll do that.

Reply:
Done